lib/ansiblelint/formatters/__init__.py

```diff
@@ -87,6 +87,33 @@ def format(self, match, colored=False):
                                     match.message)
 
 
+class AnnotationsFormatter(BaseFormatter):
+    # https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-a-warning-message
+    """Emits errors in format that is recognized by Github Annotation parsing.
+
+    ::warning file={name},line={line},col={col}::{message}
+
+    Supported levels: debug, warning, error
+    """
+
+    def format(self, match, colored=False):
+        formatstr = u"::{0} file={1},line={2}::[{3}] {4}"
+        return formatstr.format(
+            self.severity_to_level(match.rule.severity),
+            self._format_path(match.filename),
+            match.linenumber,
+            "E" + match.rule.id,
+            match.message)
```

Comment:
Let's use named placeholders, numbers in templates are confusing.
```suggestion
        workflow_command = self.severity_to_level(match.rule.severity)
        file_path = self._format_path(match.filename)
        line_num = match.linenumber
        rule_id = match.rule.id
        violation_details = match.message

        return (
            f"::{workflow_command} file={file_path},line={line_num}"
            f"::[E{rule_id}] {violation_details}"
        )
```